framework/src/modules/sequence/sequence_module.ts

```diff
@@ -29,4 +32,54 @@ export class SequenceModule extends BaseModule {
 			balance: BigInt(0),
 		},
 	};
+
+	public async beforeTransactionApply(input: TransactionApplyInput): Promise<void> {
+		const { tx, stateStore } = input;
+		const senderAddress = getAddressFromPublicKey(tx.senderPublicKey);
+		const senderAccount = await stateStore.account.getOrDefault<{ sequence: { nonce: bigint } }>(
+			senderAddress,
+		);
+
+		// Throw error when tx nonce is lower than the account nonce
+		if (tx.nonce < senderAccount.sequence.nonce) {
+			throw new SequenceModuleError(
+				`Incompatible transaction nonce for account: ${senderAccount.address.toString(
+					'base64',
+				)}, Tx Nonce: ${tx.nonce.toString()}, Account Nonce: ${senderAccount.sequence.nonce.toString()}`,
+				this.name,
+				tx.id,
+				'.nonce',
+				tx.nonce.toString(),
+				senderAccount.sequence.nonce.toString(),
+			);
+		}
+	}
+
+	public async afterTransactionApply(input: TransactionApplyInput): Promise<void> {
+		const { tx, stateStore } = input;
+		const senderAddress = getAddressFromPublicKey(tx.senderPublicKey);
+		const senderAccount = await stateStore.account.getOrDefault<{
+			sequence: { address: Buffer; nonce: bigint };
+		}>(senderAddress);
+
+		// Throw error when tx nonce is not equal to account nonce
+		if (tx.nonce !== senderAccount.sequence.nonce) {
+			throw new SequenceModuleError(
```

Comment:
it should throw the NonceOutOfRange error to be handled specifically in the transaction pool